azure/nic.sh

```diff
@@ -0,0 +1,23 @@
+# Network Interface Controller related functions
+
+fn azure_nic_create(name, group, location, vnet, subnet, securitygroup, ipforwarding) {
```

Comment:
too much arguments. Best to create an object.

```sh
nic <= azure_nic_new("nic-example", "klb-tests", "ueast")
nic <= azure_nic_set_vnet($nic, "example-vnet")
nic <= azure_nic_set_subnet($nic, "example-subnet")
nic <= azure_nic_set_secgrp($nic, "example-secgrp")
azure_nic_create($nic)
```

Because the `nic` has too many options:

```
azure network nic create -h
help:    Create a network interface
help:    
help:    Usage: network nic create [options] <resource-group> <name> <location>
help:    
help:    Options:
help:      -h, --help                                                       output usage information
help:      -v, --verbose                                                    use verbose output
help:      -vv                                                              more verbose with debug output
help:      --json                                                           use json output
help:      -g, --resource-group <resource-group>                            the name of the resource group
help:      -n, --name <name>                                                the name of the network interface
help:      -l, --location <location>                                        the location
help:      -c, --ip-config-name <ip-config-name>                            the name of the default ip config configuration
help:      -u, --subnet-id <subnet-id>                                      the subnet identifier.
help:           e.g. /subscriptions/<subscription-id>/resourceGroups/<resource-group-name>/providers/Microsoft.Network/VirtualNetworks/<vnet-name>/subnets/<subnet-name>
help:      -k, --subnet-name <subnet-name>                                  the subnet name
help:      -m, --subnet-vnet-name <subnet-vnet-name>                        the vnet name under which subnet-name exists
help:      -w, --network-security-group-id <network-security-group-id>      the network security group identifier.
help:           e.g. /subscriptions/<subscription-id>/resourceGroups/<resource-group-name>/providers/Microsoft.Network/networkSecurityGroups/<nsg-name>
help:      -o, --network-security-group-name <network-security-group-name>  the network security group name.
help:           This network security group must exist in the same resource group as the nic.
help:           Please use network-security-group-id if that is not the case.
help:      -i, --public-ip-id <public-ip-id>                                the public ip identifier.
help:           e.g. /subscriptions/<subscription-id>/resourceGroups/<resource-group-name>/providers/Microsoft.Network/publicIPAddresses/<public-ip-name>
help:      -p, --public-ip-name <public-ip-name>                            the public ip name.
help:           This public ip must exist in the same resource group as the nic.
help:           Please use public-ip-id if that is not the case.
help:      -d, --lb-address-pool-ids <lb-address-pool-ids>                  the comma separated list of load balancer address pool identifiers
help:           e.g. /subscriptions/<subscription-id>/resourceGroups/<resource-group-name>/providers/Microsoft.Network/loadbalancers/<lb-name>/backendAddressPools/<address-pool-name>
help:      -e, --lb-inbound-nat-rule-ids <lb-inbound-nat-rule-ids>          the comma separated list of load balancer inbound NAT rule identifiers
help:           e.g. /subscriptions/<subscription-id>/resourceGroups/<resource-group-name>/providers/Microsoft.Network/loadbalancers/<lb-name>/inboundNatRules/<nat-rule-name>
help:      -a, --private-ip-address <private-ip-address>                    the private ip address
help:      -b, --private-ip-version <private-ip-version>                    the private ip version, valid values are
help:           [IPv4,IPv6], default is IPv4
help:      -r, --internal-dns-name-label <internal-dns-name-label>          the internal DNS name label
help:      -f, --enable-ip-forwarding <enable-ip-forwarding>                enable ip forwarding [true, false]
help:      -t, --tags <tags>                                                the list of tags.
help:           Can be multiple. In the format of "name=value".
help:           Name is required and value is optional.
help:           For example, -t "tag1=value1;tag2"
help:      -s, --subscription <subscription>                                the subscription identifier
help:    
help:    Current Mode: arm (Azure Resource Management)
```